### ISSUE TITLE
Learnyounode  - Make it modular bug ???

### ISSUE THREAD

**Issue by dreaddy** [nodeschool/discussions#1459]
Has anyone found a solution for this?
Or 
Please tell me the tutorial is broken so I can move on.

I don't get it.

TL;DR
module.exports.amodule  - will result in data being returned. Still shows an error. 
vs
module.exports = function - the recommended way doesn't return data.

Running on windows...I think the latest node

main file:

        var mm = require("./amodule.js");
        mm.amodule(process.argv[2], process.argv[3], function(err, data){
            	console.log(data);
        });

module done the WRONG way:

    var fs = require("fs");
    var path = require("path");
    var filtered = [];

    module.exports.amodule = function(dir, ext, callback){	
    	fs.readdir(dir, function(err, list){		
    		if(err){return callback(err, null);	}
		
    		for(var x = 0; x < list.length -  1; x++ ){				
    			if( path.extname( list[x] ).match(ext) ) {					
    				filtered.push(list[x]);
    			} 			}
    		callback(null, filtered );
    	});		
    };

results in:
Your submission results compared to the expected:

                     ACTUAL                                 EXPECTED
    ────────────────────────────────────────────────────────────────────────────────
    
       "CHANGELOG.md"                      ==    "CHANGELOG.md"
       "LICENCE.md"                        ==    "LICENCE.md"
       "README.md"                         ==    "README.md"
       ""                                  ==    ""

    ────────────────────────────────────────────────────────────────────────────────

    √ Submission results match expected
    ? Your additional module file [amodule.js] does not export a single function. You must use the             `module.exports = function () {}` pattern.

Module done the recommended way

    var fs = require("fs");
    var path = require("path");
    var filtered = [];

    module.exports = function(dir, ext, callback){
	
    	fs.readdir(dir, function(err, list){
		
    		if(err){return callback(err, null);	}		
		
    		for(var x = 0; x < list.length -  1; x++ ){	
			
    			if( path.extname( list[x] ).match(ext) ) {		
				
    				filtered.push(list[x]);
    			} 		
    		}
    		callback(null, filtered );
    	});		
    };

results in:
Your submission results compared to the expected:

                     ACTUAL                                 EXPECTED
    ────────────────────────────────────────────────────────────────────────────────

       ""                                  !=    "CHANGELOG.md"
                                           !=    "LICENCE.md"
                                           !=    "README.md"
                                           !=    ""
    
    ────────────────────────────────────────────────────────────────────────────────

    ? Submission results did not match expected!
    √ Additional module file exports a single function
    √ Additional module file exports a function that takes 3 arguments
    √ Additional module file handles errors properly
    √ Additional module file handles callback argument
    √ Additional module file returned two arguments on the callback function
    ? Your additional module file [amodule.js] did not return an Array object as the second argument of     the callback
    √ Additional module file returned two arguments on the callback function
    ? Your additional module file [amodule.js] did not return an Array object as the second argument of the callback

**Comment by dreaddy** [comment]
Non issue. Dammit. the main file should be.

    var amodule = require("./amodule");

    amodule(process.argv[2], process.argv[3], function(err, data){
    	
    	data.forEach(function(entry) {
    	    console.log(entry);
    	});	
    });

and the recommended method is correct.

Apologies.
